### ISSUE TITLE
Some references missing in extraction

### ISSUE THREAD

**Issue by AbhaMoitra** [GEGlobalResearch/DARPA-ASKE-TA1#134]
When ingesting the attached file (rename it by stripping out suffix '.txt'), the resulting extraction has 2 references for the variable currentTime. There should be 3 references for it. This was verified by @agabaldon and he is looking into it.

[StrippedSimplyPID.java.txt](https://github.com/GEGlobalResearch/DARPA-ASKE-TA1/files/5770343/StrippedSimplyPID.java.txt)

**Comment by crapo** [comment]
If I were going to guess, I would expect that the JavaModelExtractorJP isn't handling the expression

`final double dt = (previousTime != Double.NaN) ? (double)(currentTime - previousTime) : 0;`

which probably isn't one we encountered in our NASA Java Applet or WindTurbine examples.

**Comment by agabaldon** [comment]
@crapo, yes. For statements of type VariableDeclarationExpr, the RHS should be treated similar to RHS of statements of type AssignExpr.  There is also no case for processing conditional expressions. I will push the change to handle the RHS of VariableDeclarationExpr for your review.

**Comment by agabaldon** [comment]
Pushed to CodeExtraction branch.

**Comment by agabaldon** [comment]
Also pushed handling of Conditional and Cast expressions.
